pkg/server/render.go

```diff
@@ -59,19 +61,53 @@ func (ctrl *Controller) renderHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (ctrl *Controller) renderDiffHandler(w http.ResponseWriter, r *http.Request) {
-	var p renderParams
-	if err := ctrl.renderParametersFromRequest(r, &p); err != nil {
-		ctrl.writeInvalidParameterError(w, err)
-		return
-	}
-	if ok := ctrl.expectJSON(w, p.format); !ok {
-		return
-	}
 
-	leftStartTime, leftEndTime, leftOK := parseRenderRangeParams(r.URL.Query(), "leftFrom", "leftUntil")
-	rghtStartTime, rghtEndTime, rghtOK := parseRenderRangeParams(r.URL.Query(), "rightFrom", "rightUntil")
-	if !leftOK || !rghtOK {
-		ctrl.writeInvalidParameterError(w, errTimeParamsAreRequired)
+	var (
+		p             renderParams
+		rP            RenderDiffParams
+		leftStartTime time.Time
+		leftEndTime   time.Time
+		rghtStartTime time.Time
+		rghtEndTime   time.Time
+
+		leftOK bool
+		rghtOK bool
+	)
+
+	if r.Method == "GET" {
```

Comment:
What do you think about introducing a function that would take care of diff parameters regardless of the method, something like this:
```
func (ctrl *Controller) renderDiffParametersFromRequest(r *http.Request, p *RenderDiffParams) error
```

Reply:
I think since we would be deprecating the GET method, I would like to keep the request struct for GET and POST different, so I am using different methods.
Let me know if this makes sense 😄

Reply:
Huh, that's right :) Let's leave it as it is